### ISSUE TITLE
Doesn't work on Windows because tar is not available

### ISSUE THREAD

**Issue by ttonyh** [swarajban/npm-cache#4]
Windows doesn't have a tar exe so an error is thrown. This module is very useful and it would be nice to see it work in Windows.

**Comment by swarajban** [comment]
Ah, haven't tried on Windows. Are you running via cygwin or Command Prompt?

**Comment by ttonyh** [comment]
I'm running in Command Prompt; I assume it would work in cygwin, but that would require another dependency.

**Comment by swarajban** [comment]
Ok, I'll try to get a fix out this week

**Comment by rasmus** [comment]
+1

**Comment by swarajban** [comment]
Still working on this btw. I have a version in the `develop` branch that uses a cross-platform compatible `tar gz`, but there are still a handful of shell commands I need to port over.

**Comment by ttrentham** [comment]
I'd like to understand what your workaround will be. Any ETA? I'll take a look at the develop branch. We ran into this issue today as we just implemented npm-cache. It works fine for those developing on OSX and Linux, but our Windows developers hit this issue.

**Comment by swarajban** [comment]
HI @ttrentham sorry it's been so long, been swamped at work. The main reason the package isn't working on windows is the `tar` command isn't available in windows command prompt or cygwin (in many  installations). There is some in-progress code in the develop branch that uses a supposed x-platform tool for tar/gzipping files that should fix the issues on windows. 

I don't wanna promise any ETA on final released solution bc I'm fairly busy these days, but I hope this helps.

**Comment by ttrentham** [comment]
Thanks for the response. I appreciate the work. Is there something in particular holding you up from rolling it out? Perhaps I or others could help?

**Comment by swarajban** [comment]
Main issue for me is time, doubt there are any actual showstoppers. Would be happy to accept a PR fo rthis

**Comment by shaneray** [comment]
+1

**Comment by shaneray** [comment]
I tried to pull down the develop branch and use it on Windows.  It seemed to be working at first but then I got the following error.

`'rm' is not recognized as an internal or external command`

This seemed to be when it was calling the fs.unlink command.  I am not at node expert at all, but I wonder if it may be a bug in the fs library.  Maybe a simple version update to this library will fix the issue.  I will try to work with it more in a bit, but wanted to put this information out there incase I did not have a chance to get to it before someone else.
